samples/logging/prometheus-grafana-opentelemetry/Core_8/Endpoint/LoadSimulator.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using NServiceBus;
+
+// Simulates busy (almost no delay) / quite time in a sine wave
+class LoadSimulator
+{
+    public LoadSimulator(IEndpointInstance endpointInstance, TimeSpan minimumDelay, TimeSpan idleDuration)
+    {
+        this.endpointInstance = endpointInstance;
+        this.minimumDelay = minimumDelay;
+        this.idleDuration = TimeSpan.FromTicks(idleDuration.Ticks / 2);
+    }
+
+    public void Start()
+    {
+        _ = Task.Run(Loop, CancellationToken.None);
```

Comment:
why not also pass the tokenSource token here?

Reply:
Updated